[RFT] lantiq: split base-files into subtargets and polish them

adschm opened the pull request:
This splits the base-files on lantiq target across base-files and uses this to update some files to be in line with modern ath79 and modernized ramips.

Please comment and test.

@hauke @dangowrt

adschm committed:
lantiq: split base-files into subtargets

This splits the device-dependent base-files into subtarget directories,
like done recently for ath79 and ramips. While this increases the
overall lines of codes, it will make the code per subtarget smaller
and easier to keep track of features and devices.

While at it, several variables at the top of 02_network are removed,
as they were never changed. The values are put directly into the
function calls where they are used.

Remove unneeded LED setup from 01_leds, and remove 01_leds entirely
for falcon subtarget (as it is not used there).

Applies alphabetic reordering to device cases in base-files.

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>

adschm committed:
lantiq: reorganize 02_network board.d files

This reorganizes 02_network board.d files based on what's done for
ath79 and ramips: Instead of putting all settings into a single big
case, the interface/dsl/MAC address setup is put into separate
functions with a specific switch case for each of them. This makes
grouping of devices much easier and should be easier to read, too.

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>

adrianschmutzler commented on target/linux/lantiq/ase/base-files/etc/board.d/02_network:
```diff
@@ -35,7 +34,7 @@ esac
 	ucidef_add_atm_bridge 1 32 "llc" "bridged" "dsl"
 
-if lantiq_is_vdsl_system; then
+if grep -qE "system type.*: (VR9|xRX200)" /proc/cpuinfo; then
```
This looks to me like it could be subtarget-specific, and thus the if might be removed with subtarget-specific files now?
Can somebody assist me on that one?

adrianschmutzler commented:
Size decrease is not impressive, however IMO this is more about maintainability anyway:

OLD $ ls -al bin/targets/lantiq/*/packages/base-files*
-rw-r--r-- 1 adsc adsc 50630 Dec 17 19:20 bin/targets/lantiq/ase/packages/base-files_210-r0+11685-f45a16dead_mips_mips32.ipk
-rw-r--r-- 1 adsc adsc 52762 Dec 17 19:19 bin/targets/lantiq/falcon/packages/base-files_210-r0+11685-f45a16dead_mips_24kc.ipk
-rw-r--r-- 1 adsc adsc 52760 Dec 17 19:14 bin/targets/lantiq/xrx200/packages/base-files_210-r0+11685-f45a16dead_mips_24kc.ipk
-rw-r--r-- 1 adsc adsc 50638 Dec 17 19:17 bin/targets/lantiq/xway_legacy/packages/base-files_210-r0+11685-f45a16dead_mips_24kc.ipk
-rw-r--r-- 1 adsc adsc 52764 Dec 17 19:16 bin/targets/lantiq/xway/packages/base-files_210-r0+11685-f45a16dead_mips_24kc.ipk

NEW $ ls -al bin/targets/lantiq/*/packages/base-files*
-rw-r--r-- 1 adsc adsc 48292 Dec 17 18:01 bin/targets/lantiq/ase/packages/base-files_210-r0+11690-931cb40018_mips_mips32.ipk
-rw-r--r-- 1 adsc adsc 50358 Dec 17 18:30 bin/targets/lantiq/falcon/packages/base-files_210-r0+11690-931cb40018_mips_24kc.ipk
-rw-r--r-- 1 adsc adsc 51841 Dec 17 18:53 bin/targets/lantiq/xrx200/packages/base-files_210-r0+11690-931cb40018_mips_24kc.ipk
-rw-r--r-- 1 adsc adsc 48312 Dec 17 18:38 bin/targets/lantiq/xway_legacy/packages/base-files_210-r0+11690-931cb40018_mips_24kc.ipk
-rw-r--r-- 1 adsc adsc 51369 Dec 17 18:45 bin/targets/lantiq/xway/packages/base-files_210-r0+11690-931cb40018_mips_24kc.ipk

hauke commented on target/linux/lantiq/xway/base-files/etc/board.d/02_network:
```diff
@@ -0,0 +1,126 @@
+	ls /lib/modules/$(uname -r)/ltq_atm* 1> /dev/null 2>&1 && \
+		ucidef_add_atm_bridge 1 32 "llc" "bridged" "dsl"
+
+	if grep -qE "system type.*: (VR9|xRX200)" /proc/cpuinfo; then
```
This should never be true for the xway target which does not support the xrx200

adrianschmutzler replied:
@hauke What about xway_legacy, amazonse, falcon?

hauke reviewed (commented):
looks good in general just some small comments.

adrianschmutzler commented on target/linux/lantiq/xway/base-files/etc/board.d/01_leds:
```diff
@@ -0,0 +1,44 @@
+
+board_config_update
+
+led_wifi="$(get_dt_led wifi)"
```
I asked myself the same question and still are not sure what's the best way to do this. The shared part of the file represents a considerable part of it, so one could just have 01_leds not split across subtargets or at least create a shared file to-be-included.
On the other hand, having 01_leds in the subtargets will be tidier as we do not duplicate the board.d folder then; despite, all the duplicate lines are relatively simple ones.
I followed my weak preference for the latter approach, which will also allow to remove certain part for subtarget as suggested in your previous comment.

Plonkbong commented:
Did you have nothing better to do ? Or solve this activity any problem, or do somthing better ?
Seriosly i am afraid that people working on this great plattform, and i thing it is generall usefull, 
but it breaks my PR: https://github.com/openwrt/openwrt/pull/2531 complettly all few months i have to make my PR compatible with the current openwrt again and again.
It will not so bad if i can see it goes forward, but i see nothing.
It will be fair if you consider my PR first.

An other usefull thing is to use the right ARCH (mips_34kc instead mips_24kc) for xrx200.

adrianschmutzler commented:
@Plonkbong 

> Did you have nothing better to do ?

You are quite aggressive considered that your PR is still struggling from formal issues and hasn't yet received any comment about its content.

> Or solve this activity any problem, or do somthing better ?

Yes, it does. Keeping stuff organized helps people who add contributions and those who have to review/understand them. Having a sensible and comprehensive structure is as much of a feature as the content itself.

> Seriosly i am afraid that people working on this great plattform, and i thing it is generall usefull,
but it breaks my PR: #2531 complettly all few months i have to make my PR compatible with the current openwrt again and again.

While I'm not quite sure what you wanted to express with this sentence, note that the impact on your patch is not severe. On a quick look, it looks to me like a 5-minute rebase.

> It will not so bad if i can see it goes forward, but i see nothing.
> It will be fair if you consider my PR first.

I considered it, but your PR is huge, it adds lots of scripts and patches, and nobody reviewed it yet. So, I cannot base my stuff on it, as it's not ready for merge. Despite, I will have to wait for testers myself before I can merge this. And, like I said before, rebasing your patch won't be that hard, effectively my patchset might even help to make your contribution easier to review.

> An other usefull thing is to use the right ARCH (mips_34kc instead mips_24kc) for xrx200.

This is not how a project like this works. Developers, with and without commit rights, contribute to this project in their free time. This essentially means that they do _what_ they want _when_ they want to do it. It's not like when you are paid for work, where the people giving you money will tell you what to do.
The same applies to reviewers: Different people may have different skill sets (I wouldn't be able to review certain parts of your PR), interests (device support vs. packages vs. build tools etc.) and motivation. They will act based on that, and that's okay. So, maybe your PR is too special or too big, and there are few people being able to review it. Maybe there are not many people interested in doing lantiq device support around. Or maybe, there is just nobody motivated to review a huge device support PR at the moment.

You can address points 1 and 2 by making the job for reviewers easier: split the patch down to logical units, if possible, and provide explanation on what you do and why in the commit message, etc.

Point 3 is harder to address: You may ask nicely, and try to have no formal issues repelling a reviewer right at the beginning.
What definitely won't help you though is offending the only person ever having invested time in giving feedback on your patch so far.

adrianschmutzler commented:
Merged the patches for led-migration and the unused-include-removal, rebased the rest.

adschm committed:
lantiq: move common DSL setup into lantiq.sh

DSL setup consists of the same commands for all subtargets, so move it
into a helper function.

While at it, remove shebang from library file.

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>

adrianschmutzler commented:
Merged an updated version.
